src/test/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsTest.java

```diff
@@ -0,0 +1,145 @@
+
+    @Mock
+    private CISShadowMonitor mockShadowMonitor;
+
+    @Mock
+    private SecurityService mockSecurityService;
+
+    @Mock
+    private GreengrassServiceClientFactory mockClientFactory;
+
+    @TempDir
+    Path tmpPath;
+
+    private CertificatesConfig certificatesConfig;
+    private Topics configurationTopics;
+    private CertificateManager certificateManager;
+    private CertificateStore certificateStore;
+    private ClientDeviceAuthMetrics metrics;
+    private CertificateSubscriptionHandler certificateSubscriptionHandler;
+    private CertificateRotationHandler certRotationMonitor;
+    private Clock clock;
+
+    @BeforeEach
+    void beforeEach() {
+        clock = Clock.fixed(Instant.now(), ZoneId.systemDefault());
+        DomainEvents domainEvents = new DomainEvents();
+        metrics = new ClientDeviceAuthMetrics(clock);
+        certificateSubscriptionHandler = new CertificateSubscriptionHandler(domainEvents, metrics);
+        certificateSubscriptionHandler.listen();
+        certificateStore = new CertificateStore(tmpPath, domainEvents, mockSecurityService);
+        certRotationMonitor = new CertificateRotationHandler(mockConnectivityInformation, domainEvents);
+
+        certificateManager = new CertificateManager(certificateStore, mockConnectivityInformation,
+                mockCertExpiryMonitor, mockShadowMonitor, Clock.systemUTC(), mockClientFactory, mockSecurityService,
+                certRotationMonitor, domainEvents);
+
+        configurationTopics = Topics.of(new Context(), KernelConfigResolver.CONFIGURATION_CONFIG_KEY, null);
+        certificatesConfig = new CertificatesConfig(configurationTopics);
+
+        certificateManager.updateCertificatesConfiguration(certificatesConfig);
+    }
+
+    @AfterEach
+    void afterEach() throws IOException {
+        configurationTopics.getContext().close();
+    }
+
+    @Test
+    void GIVEN_defaultCertManager_WHEN_subscribeToCertificateUpdates_THEN_subscribeSuccessMetricCorrectlyEmitted()
+            throws KeyStoreException, CertificateGenerationException {
+        certificateManager.generateCA("", CertificateStore.CAType.RSA_2048);
+
+        Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> con = TestUtils.asyncAssertOnConsumer((a) -> {
+        }, 3);
+
+        GetCertificateRequestOptions requestOptions = new GetCertificateRequestOptions();
+        requestOptions.setCertificateType(GetCertificateRequestOptions.CertificateType.SERVER);
+        GetCertificateRequest certificateRequest =
+                new GetCertificateRequest("testService", requestOptions, con.getRight());
+
+        // Subscribe multiple times to ensure the Cert.SubscribeSuccess metric increments correctly
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
```

Comment:
Ok, everything else looks good, but this can be improved still :) 

The beautiful thing about using events is that you don't need to deal with the certificate manager at all to test it. All you need to do is emit a CertificateSubscriptionEvent. You can get rid of all of the mocks you're using above that you need to instantiate the cert manager. Right now this test is fragile. If we make changes in the certificate manager, we wouldn't want metric tests to break. 

Later we can add an integration test to make sure that the end to end flow works as expected. But that test doesn't need to be aware of CDA internals. They rely only on external interfaces, so they won't break if we change domain internals